src/hotspot/share/opto/lcm.cpp

```diff
@@ -76,7 +76,8 @@ static bool needs_explicit_null_check_for_read(Node *val) {
   return true;
 }
 
-void PhaseCFG::move_into(Node* n, Block* b) {
+void PhaseCFG::move_node_and_its_projections_to_block(Node* n, Block* b) {
+  assert(n->bottom_type() != Type::CONTROL, "cannot move control node");
```

Comment:
I usually check `n->is_CFG()`.

What is the bottom type of an `IfNode`?
`virtual const Type *bottom_type() const { return TypeTuple::IFBOTH; }`
Are you aware of that?

Reply:
Ah, makes sense, did not know that :blush: Thanks for the update!